cloud/amazon/ec2_group_facts.py

```diff
@@ -0,0 +1,186 @@
+'''
+
+EXAMPLES = '''
+- name: Find all EC2 security groups in my VPC
+  ec2_group_facts:
+    vpc_id: vpc-b4f0exxx
+    region: us-east-1
+  register: ec2_groups_found
+
+- name: Find all EC2 security groups with name database_server
+  ec2_group_facts:
+    name: database_server
+  register: ec2_groups_found
+
+- name: Find all EC2 security groups with description my database sec group
+  ec2_group_facts:
+    description: 'my database sec group'
+  register: ec2_groups_found
+
+- debug: var=ec2_groups_found
+'''
+
+RETURN = '''
+group_name:
+    description: Name of the security group.
+    returned: when group is found
+    type: string
+    sample: "database_server"
+group_id:
+    description: ID of the security group.
+    returned: when group is found
+    type: string
+    sample: "sg-ded1b2a7"
+ip_permissions:
+    description: List of incomming port permissions for the security group, list of dictionaries.
+    returned: when group is found
+    type: list
+    sample: "{ ip_protocol: ssh, from_port: 22, to_port: 22, ip_ranges: [], prefix_list_ids: [], user_id_group_pairs: [] }"
+ip_permissions_egress:
+    description: List of outgoing port permissions for the security group, list of dictionaries.
+    returned: when group is found
+    type: list
+    sample: "{ ip_protocol: ssh, from_port: 22, to_port: 22, ip_ranges: [], prefix_list_ids: [], user_id_group_pairs: [] }"
+vpc_id:
+    description: ID of the VPC the security group belongs to.
+    returned: when group is found
+    type: string
+    sample: "vpc-b4f0exxx"
+owner_id:
+    description: The AWS account ID of the security group owner.
+    returned: when group is found
+    type: string
+    sample: 099720109471
+description:
+    description: Description text of the security group.
+    returned: when group is found
+    type: string
+    sample: "database sec group"
+tags:
+    description: AWS tags attached to the security group, list of dictionaries.
+    returned: when group has tags
+    type: list
+    sample: { "key": "value" }
+'''
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def list_ec2_security_groups(connection, module):
+    name = module.params['name']
+    description = module.params['description']
+    vpc_id = module.params['vpc_id']
+    group_id = module.params['group_id']
+
+    # Create filter params
+    filter = {}
+
+    if name:
+        filter['group-name'] = name
+    if description:
+        filter['description'] = decription
+    if vpc_id:
+        filter['vpc-id'] = vpc_id
+    if group_id:
+        filter['group-id'] = group_id
+
+    filters = ansible_dict_to_boto3_filter_list(filter)
+
+    # find groups based on filter
+    try:
+        groups = connection.describe_security_groups(Filters=filters)
+    except ClientError as e:
+        module.fail_json(msg=e.message)
```

Comment:
Handle the exception as per the AWS guidelines

E.g.
except ClientError, e:
    module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))